### PR TITLE
feat(web): adopt browser routing for cleaner URIs

### PR DESCRIPTION
**Pull request by stevapple**
This PR changes `docat` to use browser history for routing, for simpler implementation and cleaner URIs.

**BREAKING:** `doc` and `api` are no longer valid project names.

Before:
```
https://doc.example.org/#/project/version/path%2Fto%2Fpage.html#hash
```

After:
```
https://doc.example.org/project/version/path/to/page.html#hash
```

Old-styled URIs will be automatically rewritten to newer ones upon loading.

### CODE REVIEW

**Comment by stevapple** [comment]
Mark as draft until backward compatibility is addressed.

**Comment by randombenj** [comment]
Thanks for the changes @stevapple, we just had a 'facelift' of the frontend page of the UI, would you mind rebasing onto `main` before I have a look?

**Comment by stevapple** [comment]
> Thanks for the changes @stevapple, we just had a 'facelift' of the frontend page of the UI #907, would you mind rebasing onto `main` before I have a look?

Done

File: docat/docat/utils.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def is_forbidden_project_name(name: str) -> bool:
     a page on the docat website.
     """
     name = name.lower().strip()
-    return name in ["upload", "claim", "delete", "help"]
+    return name in ["upload", "claim", "delete", "help", "doc"]
```
Comment (randombenj) [inline docat/docat/utils.py:129]: This is fine with me @fliiiix what do you think?

**Review by randombenj — APPROVED** [review]
This is awesome!

File: docat/docat/utils.py (hunk 1 above)
Comment (stevapple) [reply docat/docat/utils.py]: It seems that `api` should also be added to the list...

**Comment by stevapple** [comment]
@randombenj @fliiiix bother for a second review?

**Review by randombenj — APPROVED** [review]

**Pull request merged**